Configuration: Simplify generating list of generated files in build file templates

levitte opened the pull request:
Computing the value of the GENERATED variable in the build file
templates is somewhat overcomplicated, and because of possible
duplication errors, changes are potentially error prone.

Looking more closely at how this list is determined, it can be
observed that the exact list of files to check is consistently
available in all the values found in the `%unified_info` tables
'depends', 'sources' and 'shared_sources', and all that's needed is to
filter those values so only those present as keys in the 'generate'
table are left.

This computation is also common for all build files, so due to its
apparent complexity, we move it to common0.tmpl, with the result left
in a global variable (`@generated`), to be consumed by all build file
templates.

common0.tmpl is included among the files to process when creating
build files, but unlike common.tmpl, it comes first of all.

levitte committed:
Configuration: Simplify generating list of generated files in build file templates

Computing the value of the GENERATED variable in the build file
templates is somewhat overcomplicated, and because of possible
duplication errors, changes are potentially error prone.

Looking more closely at how this list is determined, it can be
observed that the exact list of files to check is consistently
available in all the values found in the %unified_info tables
'depends', 'sources' and 'shared_sources', and all that's needed is to
filter those values so only those present as keys in the 'generate'
table are left.

This computation is also common for all build files, so due to its
apparent complexity, we move it to common0.tmpl, with the result left
in a global variable (@generated), to be consumed by all build file
templates.

common0.tmpl is included among the files to process when creating
build files, but unlike common.tmpl, it comes first of all.

levitte committed:
fixup! Configuration: Simplify generating list of generated files in build file templates

dot-asm commented:
As for red cross from travis. Looks like another intermittent impossible problem on osx [again]. Impossible becasue we know that server listens, but client fails to connect. What one can do is to arrange it so that it doesn't time out for 10 minutes...

levitte committed:
fixup! Configuration: Simplify generating list of generated files in build file templates

levitte commented:
> nmake build_all_generated says "don't know how to make 'crypto\aes\aes-x86_64.s'. In other words this doesn't work as is.

Thanks.  Fixed, commit coming up.

levitte committed:
fixup! Configuration: Simplify generating list of generated files in build file templates

levitte commented:
> On related note one can wonder if it would be sensible to exercise build_all_generated through ci? As '[n]make build_all_generated' followed by '[n]make PERL=foobar', so that we notice when it breaks...

Sounds like a good idea, yeah

dot-asm commented on Configurations/windows-makefile.tmpl:
```diff
@@ -88,12 +88,9 @@ DEPS={- join(" ", map { (my $x = $_) =~ s|\.o$|$depext|; $x; }
-                    grep { /\.o$/ } keys %{$unified_info{sources}} ),
-                  ( grep { /\.(?:h|def)$/ } keys %{$unified_info{generate}} )) -}
+GENERATED={- # common.tmpl provides @generated
+             join(" ", map { (my $x = $_) =~ s|\.[sS]$|.asm|; $x }
```
As you might have noticed (in #5929) I've started experimenting in complete .S-rule on Windows. Intention is that it should become handy on ARM. That means that .S files would have to be viewed as legitimate output for pl. Just in case for reference, I haven't modified it in #5929, because for far I was toying with already existing IA64 files. This is naturally a **side note**.

levitte replied:
I did notice your change, but was unsure where it was going, so...

levitte replied:
I'm thinking that this line will get modified when needed

levitte committed:
.travis.yml: modify to exercise generation with build_all_generated only

levitte commented:
I've modified .travis.yml to test build_all_generated.  The same should be done with appveyor.yml, but my PowerShell-fu is insufficient, so if someone can help, that would be great.

dot-asm commented:
> I've modified .travis.yml to test build_all_generated

It's going to fail because you need to add $(SCRIPTS) to build_all_generated.

dot-asm commented:
> It's going to fail because you need to add $(SCRIPTS) to build_all_generated.

Not to mention that if you shortcut PERL yourself, then you *have to* configure with no-makedepend.

dot-asm commented:
PERL=no-perl is probably better choice...

Commit:
appveyor.yml: exercise build_all_generated.

levitte committed:
fixup! Configuration: Simplify generating list of generated files in build file templates

dot-asm commented:
> build_all_generated ... appveyor.yml

I've tossed a commit to #5929... [Recall that we always configure with no-makedepend in appveyor.]

levitte committed:
fixup! .travis.yml: modify to exercise generation with build_all_generated only

levitte commented:
> > I've modified .travis.yml to test build_all_generated
> 
> It's going to fail because you need to add $(SCRIPTS) to build_all_generated.

I changed the computation of `@generated` to include them...

levitte commented:
> Not to mention that if you shortcut PERL yourself, then you *have to* configure with no-makedepend.

Thanks.  Fixed

levitte commented:
> I've tossed a commit to #5929... [Recall that we always configure with no-makedepend in appveyor.]

Cheery-picked it

dot-asm commented on Configurations/unix-Makefile.tmpl:
```diff
@@ -106,12 +106,9 @@ DEPS={- join(" ", map { (my $x = $_) =~ s|\.o$|$depext|; $x; }
-                    grep { /\.(?:o|res)$/ } keys %{$unified_info{sources}} ),
-                  ( grep { /\.(?:h|map|def)$/ } keys %{$unified_info{generate}} )) -}
+GENERATED_MANDATORY={- join(" ", @{$unified_info{depends}->{""}}) -}
+GENERATED={- # common.tmpl provides @generated
```
"common0.tmpl", right?

dot-asm replied:
Multiple occurrences...

levitte replied:
Fixed everywhere

levitte committed:
fixup! Configuration: Simplify generating list of generated files in build file templates

levitte committed:
fixup! Configuration: Simplify generating list of generated files in build file templates

levitte commented:
Merged.

ad5c205c011d1e2ec538d1f4d75477133e0dcab3 appveyor.yml: exercise build_all_generated.
6197bc7c5490d215f6abc20afa27308d5e301df2 .travis.yml: exercise build_all_generated
8258975c94398930e7b5406b8a3af53a662d1354 Configuration: Simplify generating list of generated files in build file templates
